### DISCUSSION TITLE
Can I check If Object logging is enabled on s3 buckets ?

### DISCUSSION THREAD

**Discussion by akshayyeluru** [cloud-custodian/cloud-custodian#7322] [General]
I am trying to write a policy that checks if
1.Encryption is enabled on S3 buckets
2.Bucket is publicly accessible and
**3.Object level logging is enabled**
 I've done the 1st and 2nd conditions using "_no-encryption-statement_" and "_global-grants_" filters respectively, But I could not find a way to achieve the 3rd one.

Can some expert help me with this?

**Comment by kapilt** [comment]
object level logging can mean either s3 access Logs, which is added to the bucket under Logging key (try a resource policy with no filters to see all the keys that can be filtered), or it can mean data events (cloudtrail object logging) for which we have a dedicated filter data-events, although I'll note that right after we added that cloud trail s3 data-events grew an ability to log all buckets.

**Comment by akshayyeluru** [comment]
Yes I mean logging for data events (cloudtrail object logging) and accordnig to your suggestion does the following code works?

```
  - name: s3-enable-data-events-logging
    resource: s3
    filters:
     -  type: data-events
        state: absent
```
